types/model.d.ts

```diff
@@ -74,9 +90,159 @@ export interface ModelProperties {
  * }
  * ```
  */
-export interface ModelSettings extends AnyObject {
+export interface ModelSettings extends AnyObject, ModelUtilsOptions {
   strict?: boolean;
+
+  /**
+   * Set if manual assignment of auto-generated ID values should be blocked.
+   */
   forceId?: boolean;
+
+  /**
+   * @remarks
+   * Defaults to `true`.
+   */
+  idInjection?: boolean;
+  
+  plural?: string;
+
+  http?: {
+    path?: string;
+  }
```

Comment:
Deprecate; Add tsdoc explaining their purpose, defaults, and relationship between each other (e.g. `plural` is used for `http.path` when the latter is not set)